presto-product-tests/src/main/java/io/prestosql/tests/hive/TestHiveTransactionalTable.java

```diff
@@ -59,14 +66,28 @@ public void testReadFullAcid(boolean isPartitioned, BucketingType bucketingType)
             // test filtering
             assertThat(query("SELECT col, fcol FROM " + tableName + " WHERE fcol = 1 ORDER BY col")).containsOnly(row(21, 1));
 
+            // test compacted data read
+            onHive().executeQuery(
+                    "INSERT INTO TABLE " + tableName + hivePartitionString + " VALUES (20, 3)");
+            ExecutorService executor = Executors.newSingleThreadExecutor();
+            try {
+                Future result = executor.submit(() ->
+                        onHive().executeQuery("alter table " + tableName + " " + hivePartitionString + " compact 'minor' and wait"));
+                result.get(5, TimeUnit.MINUTES);
```

Comment:
Why use a Future here?

Reply:
To have an option to timeout. If hive is configured with no compactor threads, then this query will never return.

Reply:
we usually use `@Test(timeout=` for  this

Reply:
used timeout